client/v3/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2024 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"context"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// New creates KubernetesClient from config.
+// Caller is responsible to call Close() to clean up client.
+func New(cfg clientv3.Config) (*KubernetesClient, error) {
+	c, err := clientv3.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return fromClient(c, c.Close), nil
+}
+
+// FromClient creates KubernetesClient from existing client.
+// Caller can still call Close() however, it's not needed as long caller should call Close() on the original client.
+func FromClient(c *clientv3.Client) *KubernetesClient {
+	return fromClient(c, nil)
+}
+
+func fromClient(c *clientv3.Client, close func() error) *KubernetesClient {
+	return &KubernetesClient{
+		kv:    clientv3.RetryKVClient(c),
+		close: close,
+	}
+}
+
+type KubernetesClient struct {
+	kv    pb.KVClient
+	close func() error
+}
```

Comment:
Are you sure that the user (Kubernetes) will never need to call any methods which have already been supported by `clientv3.Client`? If yes, then you should return the `Interface` (of course, you need to add the `Close` method into the interface) instead of `*KubernetesClient`.
```suggestion
func fromClient(c *clientv3.Client, close func() error) *KubernetesClient {
	return &KubernetesClient{
		Client: c,
		kv:    clientv3.RetryKVClient(c),
	}
}

type KubernetesClient struct {
	*clientv3.Client
	kv    pb.KVClient
}
```

Reply:
Done